extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxImagesHelper.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Wikia\PortableInfobox\Helpers;
+
+
+class PortableInfoboxImagesHelper {
+	const MAX_DESKTOP_THUMBNAIL_HEIGHT = 500;
+
+	/**
+	 * extends image data
+	 *
+	 * @param array $data image data
+	 * @param int $width preferred thumbnail width
+	 * @return array|bool false on failure
+	 */
+	public function extendImageData( $data, $width ) {
+		global $wgPortableInfoboxCustomImageWidth;
+
+		// title param is provided through reference in WikiaFileHelper::getFileFromTitle
+		$title = $data['name'];
+		$file = \WikiaFileHelper::getFileFromTitle( $title );
+
+		if (
+			!$file || !$file->exists() ||
+			!in_array( $file->getMediaType(), [ MEDIATYPE_BITMAP, MEDIATYPE_DRAWING, MEDIATYPE_VIDEO ] )
+		) {
+			return false;
+		}
+
+		// get dimensions
+		$originalWidth = $file->getWidth();
+		$dimensions = $this->getThumbnailSizes(
+			$width, self::MAX_DESKTOP_THUMBNAIL_HEIGHT, $originalWidth, $file->getHeight() );
+		// if custom and big enough, scale thumbnail size
+		$ratio = !empty( $wgPortableInfoboxCustomImageWidth ) && $originalWidth > $wgPortableInfoboxCustomImageWidth ?
+			$wgPortableInfoboxCustomImageWidth / $dimensions['width'] : 1;
+		// get thumbnail
+		$thumbnail = $file->transform( [
+			'width' => round( $dimensions['width'] * $ratio ),
+			'height' => round( $dimensions['height'] * $ratio )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "height" 7 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)